first_order_model/train.py

```diff
@@ -201,19 +202,126 @@ def train(config, generator, discriminator, kp_detector, checkpoint, log_dir, da
     generator_full = GeneratorFullModel(kp_detector, generator, discriminator, train_params)
     discriminator_full = DiscriminatorFullModel(kp_detector, generator, discriminator, train_params)
     
+    for x in dataloader:
```

Comment:
What is the purpose of this block of code? Can you add a comment? I think it will break some of the backwards compatibility if I'm training non-netadapt. So please first clarify why it is needed, and next put protection to ensure backward compatibility